### PR TITLE
[pvr] adds new epg properties like writer, directory, cast, original title etc.

### PR DESCRIPTION
**Pull request by xhaggi**
@Angelinas1 asked me to pick-up his work from #6265 and do the necessary changes, because he was unable to re-organize the PR by him-self.

This PR adds the following new properties to an EPG item:
- original title
- cast
- directory
- writer
- year
- IMDB number

@Jalle19 or @opdenkamp mind taking a look.

### CODE REVIEW

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -5198,6 +5213,26 @@ std::string CGUIInfoManager::GetItemLabel(const CFileItem *item, int info, std::
         stereoMode = CStereoscopicsManager::Get().NormalizeStereoMode(item->GetVideoInfoTag()->m_streamDetails.GetStereoMode());
       return stereoMode;
     }
+  case LISTITEM_IMDBNUMBER:
+    {
+      if (item->HasPVRChannelInfoTag())
+      {
+        CEpgInfoTagPtr tag(item->GetPVRChannelInfoTag()->GetEPGNow());
+        if (tag)
+          return tag->IMDBNumber();
```
Comment (xhaggi) [inline xbmc/GUIInfoManager.cpp]: should this be changed to?
```
if (item->HasEPGInfoTag())
  return item->GetEPGInfoTag()->IMDBNumber()
```

File: xbmc/GUIInfoManager.cpp [hunk 2]
```diff
@@ -5198,6 +5213,26 @@ std::string CGUIInfoManager::GetItemLabel(const CFileItem *item, int info, std::
...
+      }
+      break;
+    }
+  case LISTITEM_EPISODENAME:
+    {
+      if (item->HasPVRChannelInfoTag())
+      {
+        CEpgInfoTagPtr tag(item->GetPVRChannelInfoTag()->GetEPGNow());
+        if (tag)
+          return tag->EpisodeName();
```
Comment (xhaggi) [inline xbmc/GUIInfoManager.cpp]: same here?

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (xhaggi) [inline xbmc/GUIInfoManager.cpp]: do we want to return all these new properties for the channel item too?

**Comment by Angelinas1** [comment]
@xhaggi Many thanks...

**Comment by Hedda** [comment]
Please also see the forum discussion here with some more feedback on this:

http://forum.kodi.tv/showthread.php?tid=201981

One request there is to add TV season and episode number for TV Shows.

Another request is to also add id tag number for TheTVDB.com as well.

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (opdenkamp) [inline xbmc/GUIInfoManager.cpp]: if the listitem is a channel, then you can get the currently active tag from GetEPGNow(), so this looks correct
and it's doing it for the channel item?

File: xbmc/GUIInfoManager.cpp (hunk 2 above)
Comment (opdenkamp) [inline xbmc/GUIInfoManager.cpp]: same comment ;-)

File: xbmc/epg/EpgInfoTag.cpp [hunk 3]
```diff
@@ -488,6 +512,83 @@ std::string CEpgInfoTag::Plot(bool bOverrideParental /* = false */) const
...
+void CEpgInfoTag::SetCast(const std::string &strCast)
+{
+  CSingleLock lock(m_critSection);
+  m_bChanged |= (m_strCast != strCast);
+  m_strCast = strCast;
+}
+
+std::string CEpgInfoTag::Cast(void) const
+{
+  return m_strCast;
```
Comment (opdenkamp) [inline xbmc/epg/EpgInfoTag.cpp]: don't we need a mutex lock for strings?

File: xbmc/GUIInfoManager.cpp (hunk 1 above)
Comment (xhaggi) [inline xbmc/GUIInfoManager.cpp]: if we need it for the channel item it's fine as it is, but IMO we need for the epg item too.

File: xbmc/epg/EpgInfoTag.cpp (hunk 3 above)
Comment (xhaggi) [inline xbmc/epg/EpgInfoTag.cpp]: iIm not sure if this is really needed because we do not use a lock in other InfoTag's.

**Comment by Tolriq** [comment]
Should not those new fields also exported to JSON-RPC ? (Like in PVR.Details.Broadcast and friends)

**Comment by Jalle19** [comment]
@Tolriq ideally, yes

File: xbmc/epg/EpgInfoTag.cpp (hunk 3 above)
Comment (opdenkamp) [inline xbmc/epg/EpgInfoTag.cpp]: pretty sure we do ;-)

**Comment by xhaggi** [comment]
Updated to handle more info labels for channel/epg items.
@opdenkamp mind taking a look

**Comment by xhaggi** [comment]
@opdenkamp ping

File: xbmc/epg/EpgInfoTag.cpp (hunk 3 above)
Comment (opdenkamp) [inline xbmc/epg/EpgInfoTag.cpp]: lock isn't needed indeed cause we don't change existing tags

**Comment by opdenkamp** [comment]
let's merge this with your other PR so we don't need 2 API bumps. other than that, +1

File: xbmc/epg/EpgInfoTag.cpp (hunk 3 above)
Comment (ksooo) [inline xbmc/epg/EpgInfoTag.cpp]: Are you absolutely sure that we don't change existing tags? Yesterday we were at the point that we need mutex protection for CEpgInfoTag...

Comment (opdenkamp) [inline xbmc/epg/EpgInfoTag.cpp]: we change the channel pointers and recordings, but not the contents like this stuff. we only import new tags. and I guess we should refactor at some point soon to not need the setter for channel tags either, and const a bunch of methods

Comment (ksooo) [inline xbmc/epg/EpgInfoTag.cpp]: Do you remember our discussion back some months regarding setting a "playing" flag at the epginfotag, which was the trigger for my shared pointer PR? Does not fit then, right?

Comment (opdenkamp) [inline xbmc/epg/EpgInfoTag.cpp]: This is more about pvr potentially loading after the epg db was imported. if we fix the race there, and only import the epg after channels were imported, then we no longer need the setter. It's either set in the ctor then when linked to a channel, or not when the epg is used outside the pvr context, which isn't possible yet

**Comment by xhaggi** [comment]
sure, i'll create a PR for the API bump and we merge it in the right order, so if we need to revert one of them we can do it safely and there is no need to mixing PRs.

**Comment by ronie** [comment]
can ListItem.IMDBNumber / VidePlayer.IMDBNumber be implemented in a way it also works regular Movies / TV Shows and not just EPG items?

**Comment by xhaggi** [comment]
@ronie do you mean that we set this also for movies and tv shows?

**Comment by ronie** [comment]
yup, you're introducing a new infolabel here.
it would be great if skins could use it for all kinds of content, movies, tv shows, epg items...

**Comment by xhaggi** [comment]
@ronie okay will add a separate commit

**Comment by xhaggi** [comment]
@ronie done in the separate commit

**Comment by ronie** [comment]
great, thx!

**Comment by xhaggi** [comment]
@opdenkamp any objections or something else?

**Comment by rpanchon** [comment]
Hi,
In kodi 15 are this changes?

thank you

**Comment by Jalle19** [comment]
Yes.

**Comment by rpanchon** [comment]
And, how i can to add this properties to the  skin?
I try use with: Listitem.OriginalTitle, but it does not work. What is the correct mode?
Thank you

**Comment by xhaggi** [comment]
please use our forum for support on that.

**Pull request closed without merging**